lib/resolve-config.js

```diff
@@ -44,18 +45,31 @@ module.exports = (
     assets: assets ? castArray(assets) : assets,
     milestones: milestones ? castArray(milestones) : milestones,
     successComment,
-    proxy: getProxyConfiguration(defaultedGitlabUrl, HTTP_PROXY, HTTPS_PROXY),
+    proxy: getProxyConfiguration(defaultedGitlabUrl, HTTP_PROXY, HTTPS_PROXY, NO_PROXY),
   };
 };
-
-function getProxyConfiguration(gitlabUrl, HTTP_PROXY, HTTPS_PROXY) {
+ 
+function bypassProxy(NO_PROXY, gitlabUrl) {
+  if (NO_PROXY === '*') {
+    return true;
+  }
+  const hostName = gitlabUrl.split('://')[1].split(':')[0];
+  const noProxyEntries = NO_PROXY.split(',');
+  return noProxyEntries.some((proxyConfig) => {
+    return hostName.endsWith(proxyConfig.toLowerCase().trim().replace(/^\./, ""));
+  });
+}
+function getProxyConfiguration(gitlabUrl, HTTP_PROXY, HTTPS_PROXY, NO_PROXY) {
   const sharedParameters = {
     keepAlive: true,
     keepAliveMsecs: 1000,
     maxSockets: 256,
     maxFreeSockets: 256,
     scheduling: 'lifo',
   };
+  if(bypassProxy(NO_PROXY, gitlabUrl)) {
+    return {};
+  }
```

Comment:
Nit: Maybe most this to the beginning of the function.

Reply:
Reworked on using proxy-from-env package